bundler/lib/bundler/lazy_specification.rb

```diff
@@ -20,11 +20,15 @@ def initialize(name, version, platform, source = nil)
     end
 
     def full_name
-      @full_name ||= if platform == Gem::Platform::RUBY
-        "#{@name}-#{@version}"
-      else
-        "#{@name}-#{@version}-#{platform}"
-      end
+      @full_name ||= name_tuple.full_name
```

Comment:
this will cause a name_tuple to be allocated which feels unnecessary to me

Reply:
Reverting. Should I do something els with `lock_name` below too?